WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java

```diff
@@ -84,12 +88,36 @@
     private static final DefaultActionResolver _actionResolver = new DefaultActionResolver(WNPRC_PurchasingController.class);
     public static final String NAME = "wnprc_purchasing";
     public static final Double ADDITIONAL_REVIEW_AMT = 5000.0; // additional review is required for requests >= $5000
+    private static final String FOLDER_ADMIN_ROLE = "Folder Administrator";
 
     public WNPRC_PurchasingController()
     {
         setActionResolver(_actionResolver);
     }
 
+    @NotNull
+    private List<User> getFolderAdmins()
+    {
+        List<User> adminUsers = SecurityManager.getUsersWithPermissions(getContainer(), Collections.singleton(AdminPermission.class));
+        List<RoleAssignment> folderAdminGroups = getContainer().getPolicy().getAssignments().stream().filter(roleAssignment -> roleAssignment.getRole().getName().equals(FOLDER_ADMIN_ROLE)).collect(Collectors.toList());
+        int folderAdminUserId = folderAdminGroups.get(0).getUserId();
+
+        List<User> folderAdmins = new ArrayList<>();
```

Comment:
Instead of getting the users with permissions and reconciling against the folder admin list, I recommend iterating through the role assignments and using SecurityManager.getAllGroupMembers() to expand to a user list. Please be sure to test both direct user assignments and group assignments to the role.

Reply:
Done! Thank your for suggesting this, it is much better!